src/service.js

```diff
@@ -177,7 +190,18 @@ class Service {
   }
 
   patch (id, data, params) {
-    params.query = params.query || {};
+    const query = params.query || {};
+    const patchQuery = {};
+
+    // Account for potentially modified data
```

Comment:
Maybe the better fix is to make a `find`, run the update and then another find with the list of original ids via `find({ _id: { $in: oldFindIds } })` after.